azalea-client/src/mining.rs

```diff
@@ -66,6 +76,66 @@ impl Client {
             position,
         });
     }
+
+    /// When enabled, the bot will mine any block that it is looking at if it is reachable.
+    /// By default, reachability is set to 5 blocks.
+    pub fn auto_mine(&self, enabled: bool) {
+        let mut ecs = self.ecs.lock();
+        let mut entity_mut = ecs.entity_mut(self.entity);
+
+        if enabled {
+            entity_mut.insert(AutoMine);
+        } else {
+            entity_mut.remove::<AutoMine>();
+        }
+    }
+}
+
+#[derive(Component)]
+pub struct AutoMine;
+
+fn handle_auto_mine(
+    mut query: Query<
+        (
+            &HitResultComponent,
+            &Position,
+            Entity,
+            Option<&Mining>,
+            &InventoryComponent,
+            &MineBlockPos,
+            &MineItem,
+        ),
+        (With<AutoMine>, With<Player>, With<LocalEntity>),
+    >,
+    mut start_mining_block_event_writer: EventWriter<StartMiningBlockEvent>,
+) {
+    for (
+        hit_result_component,
+        position,
+        entity,
+        mining,
+        inventory,
+        current_mining_pos,
+        current_mining_item,
+    ) in &mut query.iter_mut()
+    {
+        let block_pos = hit_result_component.block_pos;
+
+        if (mining.is_none()
+            || !is_same_mining_target(
+            block_pos,
+            inventory,
+            current_mining_pos,
+            current_mining_item,
+        ))
+            && position.distance_to(&block_pos.to_vec3_floored()) <= 7.0
```

Comment:
I think this line is unnecessary as `update_hit_result_component` already has its own reach check.